src/io.c

```diff
@@ -132,24 +132,25 @@ int opentty(char *device)
 int my_send(int fd, char *s_buf, int len)
 {
     int i;
-    char string[256];
-
+    int wr;
+    
     // Empty buffer
-    // As tcflush doesn't work correctly, we use nonblocking read
-    fcntl(fd, F_SETFL, O_NONBLOCK);
-    while (readn(fd, string, sizeof(string)) > 0) {
-    }
-    fcntl(fd, F_SETFL, !O_NONBLOCK);
-    tcflush(fd, TCIFLUSH);
+    tcflush(fd, TCIOFLUSH);
 
     // We use the socket fixed variant from socket.c
-    writen(fd, s_buf, len);
+    wr = writen(fd, s_buf, len);
     for (i = 0; i < len; i++) {
         unsigned char byte = s_buf[i] & 255;
         logIT(LOG_INFO, ">SEND: %02X", (int)byte);
     }
 
-    return 1;
+    if (wr = i) {
```

Comment:
`if (wr = i) {` is an assignment! And why compare to i? IMHO it is better to use `len`.

```suggestion
    if (wr == len) {
```

Reply:
Yep, needs to be a comparison, not an assignment. That's a typo on my part when I re-did the changes to work within GIT.  Thanks for noticing.
"len" or "i" does not make a difference. Len may be easier to understand in the future.